h2o-core/src/main/java/water/Key.java

```diff
@@ -416,22 +429,14 @@ public String valueClassSimple() {
     int len = _kb.length;
     while( --len >= 0 ) {
       char a = (char) _kb[len];
-      if (' ' <= a && a <= '#') continue;
-      // then we have $ which is not allowed
-      if ('%' <= a && a <= '~') continue;
-      // already in the one above
-      //if( 'a' <= a && a <= 'z' ) continue;
-      //if( 'A' <= a && a <= 'Z' ) continue;
-      //if( '0' <= a && a <= '9' ) continue;
-      break;
+      if (!((' ' <= a && a <= '#') || ('%' <= a && a <= '~'))) break;
     }
     if (len>=0) {
       StringBuilder sb = new StringBuilder();
       sb.append(MAGIC_CHAR);
       for( int i = 0; i <= len; ++i ) {
         byte a = _kb[i];
-        sb.append(HEX[(a >> 4) & 0x0F]);
-        sb.append(HEX[(a >> 0) & 0x0F]);
+        sb.append(hex(a >> 4)).append(hex(a));
```

Comment:
Perhaps this line can be further sped up by replacing it with `sb.append(HEXNEW, a*2, 2)`, where  `static char[] HEXNEW = "00010203......fdfeff".toCharArray()`